core/templates/dev/head/domain/utilities/BrowserCheckerService.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility service for checking web browser type.
+ */
+
+oppia.factory('BrowserCheckerService', function() {
+  var isChromium = window.chrome;
+  var winNav = window.navigator;
+  var vendorName = winNav.vendor;
+  var isOpera = winNav.userAgent.indexOf('OPR') > -1;
+  var isIEedge = winNav.userAgent.indexOf('Edge') > -1;
+  var isIOSChrome = winNav.userAgent.match('CriOS');
+
+  var _isChrome = function() {
+    // For details on the reliabiltiy of this check, see
+    // https://stackoverflow.com/questions/4565112/
+    // javascript-how-to-find-out-if-the-user-browser-is-chrome
+    if (isIOSChrome ||
```

Comment:
Maybe use an additional paren since I don't think people tend to remember whether || or && takes precedence.

Incidentally, audio doesn't work on Chromium on Ubuntu, at least on my machine. Maybe related to https://askubuntu.com/questions/761975/chromium-is-not-generating-voice

Reply:
fixed